File: problemas/iniciante/2630/2630.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

const eye = (r, g, b) => parseInt(0.3 * r + 0.59 * g + 0.11 * b);
const mean = (r, g, b) => parseInt((r + g + b) / 3);
const min = (r, g, b) => Math.min(r, Math.min(g, b));
const max = (r, g, b) => Math.max(r, Math.max(g, b));

while (lines.length) {
    let T = parseInt(lines.shift());

    for (let i = 1; i <= T; ++i) {
        let conversao = lines.shift().trim();
        let [R, G, B] = lines.shift().trim().split(' ').map(x => parseInt(x));

        let resposta = `Caso #${i}: `;
        if (conversao === "eye") {
            resposta += `${eye(R, G, B)}`;
        } else if (conversao === "mean") {
            resposta += `${mean(R, G, B)}`;
        } else if (conversao === "max") {
            resposta += `${max(R, G, B)}`;
        } else if (conversao === "min") {
            resposta += `${min(R, G, B)}`;
        }
        console.log(resposta);
    }
}

File: problemas/iniciante/2630/2630.py
def eye(r, g, b):
    return int(0.3 * r + 0.59 * g + 0.11 * b)


def mean(r, g, b):
    return int((r + g + b)//3)


while True:
    try:
        T = int(input())

        for i in range(1, T + 1):
            conversao = input()
            R, G, B = [int(x) for x in input().strip().split(' ')]

            print(f'Caso #{i}: ', end='')
            if(conversao == 'eye'):
                print(eye(R, G, B))
            elif(conversao == 'mean'):
                print(mean(R, G, B))
            elif(conversao == 'min'):
                print(min(R, G, B))
            elif(conversao == 'max'):
                print(max(R, G, B))
    except EOFError:
        break
